GitVersionTaskAssemblyInfo.g.cs Namespace GitVersionTask

Hello,

I've just upgraded to version 3.0.

Having each `GitVersionTaskAssemblyInfo` in the default assembly is very nice for us as some of our assemblies are using `InternalsVisibleTo`

I'm unsure why but during some *Rebuild All* the namespace generated is completely random and causes a compilation error but if you build again then the issue is fixed and the correct namespace appears

Are you facing the same issue?

Example of a generated namespace:
```
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by a tool.
//     GitVersion
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------

using System;
using System.Reflection;

[assembly: AssemblyVersion("0.3.0.0")]
[assembly: AssemblyFileVersion("0.3.0.0")]
[assembly: AssemblyInformationalVersion("0.3.0-unstable.690+Branch.develop.Sha.b9d132f75f2667e0ea38ec52dd6e7a2a1b1b7004")]

namespace 1agvoemt
{

    [System.Runtime.CompilerServices.CompilerGenerated]
    static class GitVersionInformation
    {
        public static string Major = "0";
        public static string Minor = "3";
        public static string Patch = "0";
        public static string PreReleaseTag = "unstable.690";
        public static string PreReleaseTagWithDash = "-unstable.690";
        public static string BuildMetaData = "690";
        public static string FullBuildMetaData = "690.Branch.develop.Sha.b9d132f75f2667e0ea38ec52dd6e7a2a1b1b7004";
        public static string MajorMinorPatch = "0.3.0";
        public static string SemVer = "0.3.0-unstable.690";
        public static string LegacySemVer = "0.3.0-unstable690";
        public static string LegacySemVerPadded = "0.3.0-unstable0690";
        public static string AssemblySemVer = "0.3.0.0";
        public static string FullSemVer = "0.3.0-unstable.690";
        public static string InformationalVersion = "0.3.0-unstable.690+Branch.develop.Sha.b9d132f75f2667e0ea38ec52dd6e7a2a1b1b7004";
        public static string BranchName = "develop";
        public static string Sha = "b9d132f75f2667e0ea38ec52dd6e7a2a1b1b7004";
        public static string NuGetVersionV2 = "0.3.0-unstable0690";
        public static string NuGetVersion = "0.3.0-unstable0690";
        public static string CommitDate = "2015-07-28";
    }

}
```

@distantcam . Quick update, I had a quick look.

To generate the namespace you are taking the name of the project without the extension and it looks like that in some scenario msbuild uses a temp name during the build (I'm unsure why yet)

see below:
```
    43>UpdateAssemblyInfo:
         INFO [07/28/15 9:21:47:09] Version not in cache. Calculating version.
         INFO [07/28/15 9:21:47:09] Project root is: C:\a\src
         INFO [07/28/15 9:21:47:10] Using latest commit on specified branch
         INFO [07/28/15 9:21:47:10] Running against branch: develop (b9d132f75f2667e0ea38ec52dd6e7a2a1b1b7004)
    42>obj\Debug\GitVersionTaskAssemblyInfo.g.cs(18,11): error CS1001: Identifier expected [C:\a\src\SolutionDir\Project.Harness\1gd4yfjc.tmp_proj]
    42>obj\Debug\GitVersionTaskAssemblyInfo.g.cs(18,12): error CS0116: A namespace cannot directly contain members such as fields or methods [C:\a\src\SolutionDir\Project.Harness\1gd4yfjc.tmp_proj]
    42>obj\Debug\GitVersionTaskAssemblyInfo.g.cs(18,11): error CS1514: { expected [C:\a\src\SolutionDir\Project.Harness\1gd4yfjc.tmp_proj]
    42>obj\Debug\GitVersionTaskAssemblyInfo.g.cs(19,1): error CS1022: Type or namespace definition, or end-of-file expected [C:\a\src\SolutionDir\Project.Harness\1gd4yfjc.tmp_proj]
```

--- a/src/GitVersionTask/AssemblyInfoBuilder/AssemblyInfoBuilder.cs
+++ b/src/GitVersionTask/AssemblyInfoBuilder/AssemblyInfoBuilder.cs
@@ -5,7 +5,7 @@ using GitVersion;
 
 public class AssemblyInfoBuilder
 {
-    public string GetAssemblyInfoText(VersionVariables vars, string assemblyName)
+    public string GetAssemblyInfoText(VersionVariables vars, string rootNamespace)
     {
         var v = vars.ToList();
 
@@ -42,7 +42,7 @@ namespace {4}
         vars.MajorMinorPatch + ".0",
         vars.InformationalVersion,
         GenerateStaticVariableMembers(v),
-        assemblyName);
+        rootNamespace);
 
         return assemblyInfo;
     }

--- a/src/GitVersionTask/AssemblyInfoBuilder/UpdateAssemblyInfo.cs
+++ b/src/GitVersionTask/AssemblyInfoBuilder/UpdateAssemblyInfo.cs
@@ -22,6 +22,9 @@
         [Required]
         public ITaskItem[] CompileFiles { get; set; }
 
+        [Required]
+        public string RootNamespace { get; set; }
+
         [Output]
         public string AssemblyInfoTempFilePath { get; set; }
 
@@ -91,7 +94,7 @@
             }
 
             var assemblyInfoBuilder = new AssemblyInfoBuilder();
-            var assemblyInfo = assemblyInfoBuilder.GetAssemblyInfoText(versionVariables, Path.GetFileNameWithoutExtension(ProjectFile));
+            var assemblyInfo = assemblyInfoBuilder.GetAssemblyInfoText(versionVariables, RootNamespace);
             File.WriteAllText(AssemblyInfoTempFilePath, assemblyInfo);
         }
     }

--- a/src/GitVersionTask/NugetAssets/GitVersionTask.targets
+++ b/src/GitVersionTask/NugetAssets/GitVersionTask.targets
@@ -25,6 +25,7 @@
       SolutionDirectory="$(SolutionDir)"
       ProjectFile="$(MSBuildProjectFullPath)"
       IntermediateOutputPath="$(IntermediateOutputPath)"
+      RootNamespace="$(RootNamespace)"
       CompileFiles ="@(Compile)">
       <Output
         TaskParameter="AssemblyInfoTempFilePath"

